lib/translator/status.cc

```diff
@@ -0,0 +1,248 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "status.h"
+
+#include "common.h"
+
+namespace translator {
+
+namespace {  // anonymous namespace for helper functions
+
+constexpr ScsiStatus kDefaultScsiStatus{
+    .status = scsi_defs::Status::kCheckCondition,
+    .sense_key = scsi_defs::SenseKey::kNoSense,
+    .asc = scsi_defs::AdditionalSenseCode::kNoAdditionalSenseInfo,
+    .ascq = scsi_defs::AdditionalSenseCodeQualifier::kNoAdditionalSenseInfo,
+};
+
+// Section 7.1
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+ScsiStatus StatusToScsi(nvme_defs::GenericCommandStatusCode sc) {
```

Comment:
It's good practice to avoid abbreviations for variable names. "Do not abbreviate by deleting letters within a word."
https://google.github.io/styleguide/cppguide.html#Naming

Let's change `sc` to `status_code`.